iot-hub/Samples/device/MessageReceiveSample/MessageReceiveSample.cs

```diff
@@ -19,75 +20,95 @@ namespace Microsoft.Azure.Devices.Client.Samples
+
             // First receive C2D messages using the polling ReceiveAsync().
-            Console.WriteLine($"\n{DateTime.Now}> Device waiting for C2D messages from the hub for {s_receiveTimeout}...");
-            Console.WriteLine($"{DateTime.Now}> Use the Azure Portal IoT Hub blade or Azure IoT Explorer to send a message to this device.");
-            await ReceiveC2dMessagesPollingAndComplete(s_receiveTimeout);
+            Console.WriteLine($"\n{DateTime.Now}> Device waiting for C2D messages from the hub...");
+            Console.WriteLine($"{DateTime.Now}> Use the Azure Portal IoT hub blade or Azure IoT Explorer to send a message to this device.");
+            await ReceiveC2dMessagesPollingAndComplete(cts.Token);
 
             // Now subscribe to receive C2D messages through a callback.
             await _deviceClient.SetReceiveMessageHandlerAsync(OnC2dMessageReceived, _deviceClient);
             Console.WriteLine($"\n{DateTime.Now}> Subscribed to receive C2D messages over callback.");
 
             // Now wait to receive C2D messages through the callback.
             // Since you are subscribed to receive messages through the callback, any call to the polling ReceiveAsync() API will now return "null".
-            Console.WriteLine($"\n{DateTime.Now}> Device waiting for C2D messages from the hub for {s_receiveTimeout}...");
+            Console.WriteLine($"\n{DateTime.Now}> Device waiting for C2D messages from the hub...");
             Console.WriteLine($"{DateTime.Now}> Use the Azure Portal IoT Hub blade or Azure IoT Explorer to send a message to this device.");
-            await Task.Delay(s_receiveTimeout);
+
+            try
+            {
+                await Task.Delay(-1, cts.Token);
+            }
+            catch (TaskCanceledException)
+            {
+                // Done running.
+            }
 
             // Now unsubscibe from receiving the callback.
-            await _deviceClient.SetReceiveMessageHandlerAsync(null, _deviceClient);
+            await _deviceClient.SetReceiveMessageHandlerAsync(null, null);
         }
 
-        private async Task ReceiveC2dMessagesPollingAndComplete(TimeSpan timeout)
+        private async Task ReceiveC2dMessagesPollingAndComplete(CancellationToken ct)
         {
-            var sw = new Stopwatch();
-            sw.Start();
+            Console.WriteLine($"{DateTime.Now}> Trying to receive C2D messages by polling using the ReceiveAsync() method. Press 'n' to move to the next phase.");
 
-            Console.WriteLine($"{DateTime.Now}> Receiving C2D messages on the polling ReceiveAsync().");
-            while (sw.Elapsed < timeout)
+            while (!ct.IsCancellationRequested)
             {
-                using Message receivedMessage = await _deviceClient.ReceiveAsync(timeout);
+                if (Console.KeyAvailable
+                    && ConsoleKey.N == Console.ReadKey().Key)
```

Comment:
is this case-sensitive?

Reply:
how to manage this "move" in the pipeline?

Reply:
The pipeline build script needs to be updated: https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/build.ps1#L115

Reply:
It is not. :)